### ISSUE TITLE
Possible to do key/value in data-list?

### ISSUE THREAD

**Issue by mrmanishs** [LeaVerou/awesomplete#16722]
Hello there, I would like to have a hidden key for the value displayed for data-list, which would be sent back as the params. Is there a way to set that up?

**Comment by Loewengyrl** [comment]
I too would be very interested in this possibility!

**Comment by LeaVerou** [comment]
This is currently possible, but would require a nontrivial amount of custom code. However, this is top priority once I find some time to make improvements to Awesomplete, as it’s one of the top requested features.

**Comment by zurnet** [comment]
Adding my vote in for this as well :+1:

**Comment by lbilharz** [comment]
:+1:

**Comment by matanox** [comment]
I think this and https://github.com/LeaVerou/awesomplete/issues/16436, https://github.com/LeaVerou/awesomplete/issues/593 are tightly related, or at least should be handled as one (IMHO).

**Comment by zurnet** [comment]
Any idea on a time frame for this feature?  Hoping maybe by Sep, it's something I'm looking forward to implementing and need that key/value option.  Thanks again for an awesome plugin.  

PS... your book is great!  Definitely learned some things, even though I've been coding CSS from the beginning.

**Comment by vlazar** [comment]
**To everyone interested in differentiating value from displayed text, aka key/value feature, aka array of objects feature.**

The code is ready and it's almost 100% backward compatible. To ensure everything working fine for you we need your help with testing it in the wild before it's released. See this PR: [Separate label/value for each suggestion in list](https://github.com/LeaVerou/awesomplete/pull/16852)

**Comment by vlazar** [comment]
Separate `label/value` is landed #16866
It is also possible to initialize list with separate `label/value` via `<datalist>` or `<ul>`.

Read about different `label` and `value` at the end of [Basic usage](http://leaverou.github.io/awesomplete/#basic-usage) section and about new `data` method at the end of [Extend](http://leaverou.github.io/awesomplete/#extensibility) section.

**Comment by ArondeParon** [comment]
Hate to dig up this old thread, but what kind of sorcery do I have to do to get this working? My HTML looks like this, but when I select a value, I see the value being displayed and not the label. The suggestions work fine, but when I select them, things break.

```
<input class="awesomplete" list="mylist" />
<datalist id="mylist">
	<option value="1">Foo</option>
	<option value="2">Bar</option>
	<option value="3">Baz</option>
</datalist>
```
